cmd/podman/machine/machine.go

```diff
@@ -57,6 +59,9 @@ func machinePreRunE(c *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
+	if provider.VMType() == define.LibKrun && runtime.GOARCH == "amd64" {
+		return errors.New("libkrun is not supported on Intel based machines. Please revert to the applehv provider")
+	}
```

Comment:
This seems to be done at the wrong place. Why bloat the generic machinePreRunE with platform specific details.
This really should be handled in provider.Get()